src/utils/validators.js

```diff
@@ -429,6 +459,112 @@ const validateSections = (sectionError, sectionType, field, value) => {
+        errorMessage = `Local numbers should start with +65 followed by 8 digits starting with 6, 8 or 9. Toll free numbers should start with 1800 followed by 7 digits.`
+      }
+      break;
+    case 'email':
+      if ( value && !emailRegexTest.test(value) ) {
+        errorMessage = `Emails should follow the format abc@def.gh and should not contain special characters such as: ?!#\\$% ` //TODO
+      }
+      break;
+    case 'other':
+      if ( value.length > CONTACT_DESCRIPTION_MAX_LENGTH ) {
+        errorMessage = `Description should be shorter than ${CONTACT_DESCRIPTION_MAX_LENGTH} characters.`;
+      }
+      break;
+  }
+  return errorMessage
+}
+
+// Locations
+
+const validateLocation = (locationType, value) => {
+  let errorMessage = '';
+  switch (locationType) {
+    case 'title':
+      // Title is too short
+      if (value.length < LOCATION_TITLE_MIN_LENGTH) {
+        errorMessage = `Title should be longer than ${LOCATION_TITLE_MIN_LENGTH} characters.`;
+      };
+      // Title is too long
+      if (value.length > LOCATION_TITLE_MAX_LENGTH) {
+        errorMessage = `Title should be shorter than ${LOCATION_TITLE_MAX_LENGTH} characters.`;
+      };
+      break;
+    case 'address':
+      let errors = [];
+      // check if in-between fields are empty e.g. field 3 is filled but field 2 is empty
+      if ( (value[2].length && !(value[0].length && value[1].length)) || (value[1].length && !value[0].length) ) {
+        errors.push('Please do not leave in-between fields empty.');
+      } else {
+        value.forEach(field => {
+          let error = ''
+          // if else check necessarily because we want to push an empty string if there's no error
+          if (field && field.length < LOCATION_ADDRESS_MIN_LENGTH) {
+            error = `Field should be longer than ${LOCATION_ADDRESS_MIN_LENGTH} characters.`;
+          } 
+          if (field && field.length > LOCATION_ADDRESS_MAX_LENGTH) {
+            error = `Field should be shorter than ${LOCATION_ADDRESS_MAX_LENGTH} characters.`;
+          }
+          errors.push(error)
+        })
+      } 
+      errorMessage = errors;
+      break;
+    // fields below are operating hours fields
+    case 'days':
+      if (value && !alphabetsRegexTest.test(value)) {
+        errorMessage += `Field should only contain alphabets. `
+      }
+      if (value && value.length < LOCATION_OPERATING_DAYS_MIN_LENGTH) {
+        errorMessage += `Field should be longer than ${LOCATION_OPERATING_DAYS_MIN_LENGTH} characters. `
+      }
+      if (value && value.length > LOCATION_OPERATING_DAYS_MAX_LENGTH) {
+        errorMessage += `Field should be shorter than ${LOCATION_OPERATING_DAYS_MAX_LENGTH} characters. `
+      }
+      break;
+    case 'time': {
+      if (value && !alphanumericRegexTest.test(value)) {
+        errorMessage += `Field should only contain alphanumeric characters. `
+      }
+      if (value && !value.length < LOCATION_OPERATING_HOURS_MIN_LENGTH) {
```

Comment:
There is a bug here, it should be `value.length < LOCATION_OPERATING_HOURS_MIN_LENGTH` instead of `!value.length < LOCATION_OPERATING_HOURS_MIN_LENGTH`

Reply:
this has been fixed in the latest branch of code https://github.com/isomerpages/isomercms-frontend/tree/feat/contact-us-page-2